x-pack/plugins/security_solution/public/management/pages/policy/view/policy_forms/protections/ransomware.tsx

```diff
@@ -0,0 +1,364 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useCallback, useMemo } from 'react';
+import { useDispatch } from 'react-redux';
+import styled from 'styled-components';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import {
+  EuiCallOut,
+  EuiCheckbox,
+  EuiRadio,
+  EuiSpacer,
+  EuiSwitch,
+  EuiText,
+  EuiTextArea,
+  htmlIdGenerator,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiIconTip,
+} from '@elastic/eui';
+import { cloneDeep } from 'lodash';
+import { APP_ID } from '../../../../../../../common/constants';
+import { SecurityPageName } from '../../../../../../app/types';
+
+import {
+  Immutable,
+  OperatingSystem,
+  ProtectionModes,
+} from '../../../../../../../common/endpoint/types';
+import { RansomwareProtectionOSes, OS } from '../../../types';
+import { ConfigForm, ConfigFormHeading } from '../../components/config_form';
+import { policyConfig } from '../../../store/policy_details/selectors';
+import { usePolicyDetailsSelector } from '../../policy_hooks';
+import { LinkToApp } from '../../../../../../common/components/endpoint/link_to_app';
+import { popupVersionsMap } from './popup_options_to_versions';
+
+const ProtectionRadioGroup = styled.div`
+  display: flex;
+  .policyDetailsProtectionRadio {
+    margin-right: ${(props) => props.theme.eui.euiSizeXXL};
+  }
+`;
+
+const OSes: Immutable<RansomwareProtectionOSes[]> = [OS.windows, OS.mac];
+const protection = 'ransomware';
+
+const ProtectionRadio = React.memo(({ id, label }: { id: ProtectionModes; label: string }) => {
+  const policyDetailsConfig = usePolicyDetailsSelector(policyConfig);
+  const dispatch = useDispatch();
```

Comment:
Just so that we get TS validation on dispatch, consider:

```suggestion
  const dispatch = useDispatch<(action: AppAction) => void>();
```